frontend/src/app/commonComponents/TextInput.jsx

```diff
@@ -3,46 +3,33 @@ import PropTypes from "prop-types";
 import classnames from "classnames";
 import useUniqueId from "../commonComponents/useUniqueIds";
 
-const TextInput = ({
-  value = "",
-  label,
-  name,
-  placeholder,
-  onChange,
-  type,
-  addClass,
-}) => {
-  let [newId] = useUniqueId("textinput", 1);
-
+const TextInput = ({ label, type = "text", ...inputProps }) => {
+  const [newId] = useUniqueId("textinput", 1);
+  const id = inputProps.id || newId;
+  const props = { id, type, ...inputProps };
```

Comment:
As a React dev it's really confusing to not have the incoming props named `props`.

Reply:
Good call. Renamed.